pkg/streamingpromql/operators/absent.go

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+// Provenance-includes-location: https://github.com/prometheus/prometheus/blob/main/promql/functions.go
+// Provenance-includes-license: Apache-2.0
+// Provenance-includes-copyright: The Prometheus Authors.
+
+package operators
+
+import (
+	"context"
+	"errors"
+
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/prometheus/prometheus/promql/parser"
+	"github.com/prometheus/prometheus/promql/parser/posrange"
+
+	"github.com/grafana/mimir/pkg/streamingpromql/limiting"
+	"github.com/grafana/mimir/pkg/streamingpromql/types"
+)
+
+// AbsentOperator is an operator that implements the absent() function.
+type AbsentOperator struct {
+	timeRange                types.QueryTimeRange
+	innerExpr                parser.Expr
+	inner                    types.InstantVectorOperator
+	expressionPosition       posrange.PositionRange
+	absentCount              int
```

Comment:
If I'm understanding your intention correctly, would `innerSeriesCount` be a clearer name?

Reply:
This is count how often the series is not there or being `absent`. But let me double check again according to the feedback in https://github.com/grafana/mimir/pull/10523#discussion_r1931695665 to ensure the correctness.